mysite/mysite/views/home.py

```diff
@@ -6,12 +6,71 @@
 """
 
 from django.views import View
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.http import Http404
+from mysite import settings
+from mysite.forms import SearchForm
+import requests
+
 
 
 class Home(View):
     """
     Process GET requests for the homepage. 
     """
-    def get(self, request):
-        return render(request, "home.html", {})
\ No newline at end of file
+    def get(self, request, action):
+        try:
+            url = settings.TMDB_MOVIE_GRID_URLS[action]
+        except KeyError: 
+            raise Http404()
+        response = requests.get(url.format(apikey=settings.TMDB_API_KEY))
+        if response.status_code != 200: 
+            return redirect('error/2')
+       
+        return render(request, 'home.html', {
+            'search_bar': SearchForm(), 
+            'titles': get_title_displays(response.json()) 
+        })
+    
+
+    """
+    Process POST requests for the homepage.
+    """
+    def post(self, request, action): 
+        if action != 'search': 
+            raise Http404()
+        form = SearchForm(request.POST)
+        if form.is_valid(): 
+            url = settings.TMDB_SEARCH_MOVIES_URL
+            response = requests.get(url.format(
+                apikey=settings.TMDB_API_KEY, 
+                query=form.cleaned_data['query']
+            ))
+            if response.status_code != 200: 
+                return redirect('error/2')
+            return render(request, 'home.html', {
+                'search_bar': form, 
+                'titles': get_title_displays(response.json()), 
+                'search': True 
+            })
+        else:
+            return redirect('error/4')
+            
+
+
+
+class TitleDisplay: 
+    def __init__(self, title): 
+        self.title_id = title['id']
+        self.title = title['title']
+        self.poster = None if not title['poster_path'] else settings.TMDB_IMAGE_URL + title['poster_path']
+
+
+"""
+Package JSON response data into TitleDisplay objects, 4 per row for the template
+"""
+def get_title_displays(response):
+    displays = list(map(TitleDisplay, response['results']))
+    return [displays[i:i+4] for i in range(0, len(displays), 4)]
```

Comment:
I am more of a fan of 5 movies per row tbh. What do the rest of you guys think?

Reply:
I'm alright with it either way. I can spin something up to make that a setting in the admin panel that we could change without pushing a code change, as long as the css is set up to adapt.

Reply:
Ok I did it